packages/ddp-client/livedata_connection.js

```diff
@@ -274,11 +274,44 @@ var Connection = function (url, options) {
     msg.support = self._supportedDDPVersions;
     self._send(msg);
 
+    //Mark non-retry calls as failed  This has to be done early as getting these methods out of the current block is
+    // pretty important to making sure that quiescence is properly calculated, as well as possibly moving on
+    // to another useful block.
+
+    // Only bother testing if there is an outstandingMethodBlock (there might not be, especially if we are connecting
+    // for the first time.
+    if (self._outstandingMethodBlocks.length > 0) {
+      // If there is an outstanding method block, we only care about the first one as that is the one
+      // that could have already sent messages with no response, that are not allowed to retry.
+      _.each(self._outstandingMethodBlocks[0].methods, function(methodInvoker) {
+        // if the message wasn't sent or it's allowed to retry, do nothing.
+        if (methodInvoker.sentMessage && methodInvoker.noRetry) {
+          // The next loop serves to get the index in the current method block of this method.
+          var currentMethodBlock = self._outstandingMethodBlocks[0].methods;
+          var loopMethod;
+          for (var i = 0; i < currentMethodBlock.length; i++) {
+            loopMethod = currentMethodBlock[i];
+            if (loopMethod.methodId === methodInvoker.methodId) {
+              break;
+            }
+          }
+          // Remove from current method block. This may leave the block empty, but we
+          // don't move on to the next block until the callback has been delivered, in
+          // _outstandingMethodFinished.
+          currentMethodBlock.splice(i, 1);
+          // make sure that the method is told that it failed.
+          methodInvoker.receiveResult(Meteor.Error(409, 'Method is non-idempotent but attempted to call a second time',
```

Comment:
Yeah, I was actually going to change that just as soon as I got home after re-reading the style guide today.  If 'invocation-failed' is used, the test should also be updated to expect that.